src/profile/hooks/useCommonEditHandling.ts

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+import to from 'await-to-js';
+
+import { EditDataValue, isNewItem } from '../helpers/editData';
+import { Action } from './useProfileDataEditor';
+import { ActionHandler } from '../components/editButtons/EditButtons';
+import { RowItemProps } from '../components/multiItemEditor/MultiItemEditor';
+
+type UseActionHandlingReturnType = {
+  currentAction: Action;
+  isEditing: boolean;
+  isNew: boolean;
+  actionHandler: ActionHandler;
+};
+
+export const useCommonEditHandling = (
+  props: RowItemProps
+): UseActionHandlingReturnType => {
+  const { data, onAction } = props;
+  const isNew = isNewItem(data);
+  const [isEditing, setEditing] = useState(isNew);
+  const [currentAction, setCurrentAction] = useState<Action>(undefined);
+  const actionHandler: ActionHandler = async (action, newValue) => {
+    if (action === 'set-primary' || action === 'remove' || action === 'save') {
+      setCurrentAction(action);
+    }
+    const [err] = await to(onAction(action, data, newValue as EditDataValue));
+    if (isNew) {
+      if (err) {
+        setCurrentAction(undefined);
```

Comment:
Should we also reject on error? Since there's no return inside this if, the actionHandler will resolve on line 32 even in the case of error?

Reply:
Awaiting function does not care what happens, because all error handling is here. Adding rejection would also require rejection handling to the awaiting function, because all promise rejections must be handled.